File: Convert Sorted Array to Binary Search Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode sortedArrayToBST(int[] num) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int lastIndex = num.length - 1;
        return sortedArrayToBST(num, 0, lastIndex);
    }
    
    
    public TreeNode sortedArrayToBST(int[] num, int start, int end)
    {
        if(start > end)
        {
            return null;
        }
        
        int mid = start + (end - start)/2;
        TreeNode root = new TreeNode(num[mid]);
        root.left = sortedArrayToBST(num, start, mid - 1);
        root.right = sortedArrayToBST(num, mid + 1, end);
        return root;
    }
    
}

File: Convert Sorted Array to Binary Search Tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode *sortedArrayToBST(vector<int> &num) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        return convert(num, 0, num.size()-1);
    }
    
    TreeNode *convert(vector<int> &num, int start, int end)
    {
        if(start>end)
        {
            return NULL;
        }
        
        int mid = (start+end)/2;
        
        TreeNode *root = new TreeNode(num[mid]);
        
        root->left = convert(num, start, mid-1);
        root->right = convert(num, mid+1, end);
        
        return root;
    }
};
